public/app/features/templating/formatVariableValue.ts

```diff
@@ -0,0 +1,50 @@
+import { formatRegistry, FormatRegistryID } from '@grafana/scenes';
+
+import { isAdHoc } from '../variables/guard';
+
+import { getVariableWrapper } from './LegacyVariableWrapper';
+
+export function formatVariableValue(value: any, format?: any, variable?: any, text?: string): string {
```

Comment:
if we're going to do the runtime checks, just use `unknown` to make sure we actually do them and not make unsafe assumptions.

Reply:
In fact, digging into this further, it looks like these arguments should just have proper types?


```ts
type VariableFormatFn = (a: any, b: any, c: any) => any;
type VariableFormat = VariableFormatFn | string;

export function formatVariableValue(
  value: string | string[] | object,
  format?: VariableFormat,
  variable?: VariableModel,
  text?: string
)
```

I think it's really important to have type safety in critical flows of Grafana, otherwise this will eventually cause an [incident ](https://raintank-corp.slack.com/archives/C04QK8N8ENT)

Reply:
@joshhunt this is not new code, I just moved this function out to a new file

Reply:
and this function has a new equivalent in the new scenes framework (That is much better typed), so did not spend time rewriting this function (just needed to break this function out to use it from another place)